platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,180 @@
+import { Schema, typeOfSchema } from '../types';
+import { LogLevelId } from './LogLevel';
+import { Appenders, AppenderConfigType } from './appenders/Appenders';
+
+/**
+ * Separator string that used within nested context name (eg. plugins::pid).
+ */
+const CONTEXT_SEPARATOR = '::';
+
+/**
+ * Name of the `root` context that should always be defined.
+ */
+const ROOT_CONTEXT_NAME = 'root';
+
+const defaultAppenders = new Map([
+  [
+    'console',
+    {
+      kind: 'console',
+      layout: {
+        kind: 'pattern',
+        highlight: true
+      }
+    }
+  ]
+]);
+
+const defaultLoggers = new Map([
+  ['root', { appenders: ['console'], level: 'info' as LogLevelId }]
+]);
+
+const createLoggerSchema = (schema: Schema) => {
+  const { literal, object, string, arrayOf, oneOf } = schema;
+
+  return object({
+    level: oneOf(
+      [
+        literal('all'),
+        literal('fatal'),
+        literal('error'),
+        literal('warn'),
+        literal('info'),
+        literal('debug'),
+        literal('trace'),
+        literal('off')
+      ],
+      {
+        defaultValue: 'info'
+      }
+    ),
+    appenders: arrayOf(string(), { defaultValue: [] })
+  });
+};
+
+const createLoggingSchema = (schema: Schema) => {
+  const { object, string, mapOf } = schema;
+
+  return object({
+    appenders: mapOf(string(), Appenders.createConfigSchema(schema), {
+      defaultValue: defaultAppenders
+    }),
+    loggers: mapOf(string(), createLoggerSchema(schema), {
+      defaultValue: defaultLoggers,
+      validate(value) {
+        const rootLogger = value.get(ROOT_CONTEXT_NAME);
+        if (!rootLogger) {
+          throw new Error(
+            `"${ROOT_CONTEXT_NAME}" logger should be configured!`
```

Comment:
Been playing around with this and I think we need to improve the UX. Let's say you're relying on the logging defaults and to debug something you just want to up the log level on your plugin. What feels right to me is being able to just:

```yaml
logging:
  loggers:
    myPlugin:
      level: debug
```

Not that fails with `"root" logger should be configured!`.

Another use case could be wanting to log everything from a plugin to a file while debugging, e.g. something like:

```yaml
logging:
  appenders:
    myPluginAppender:
      kind: 'file'
      path: './myPlugin.log'
      layout:
        kind: 'pattern'
  loggers:
    myPlugin:
      appenders:
        - myPluginAppender
      level: all
```

Here we could also just use the defaults for the rest.

Maybe instead of applying the defaults like we do now, we add the default logger and appender automatically, _unless_ you override them. What do you think?